src/Akka.Streams.Kafka/Stages/Consumers/Abstract/SingleSourceStageLogic.cs

```diff
@@ -134,23 +112,52 @@ protected virtual void StopConsumerActor()
             });
         }
 
-        private void PartitionsAssigned(IEnumerable<TopicPartition> partitions)
+        private class FlushMessagesOfRevokedPartitionsHandler : IPartitionEventHandler
```

Comment:
Custom `IPartitionEventHandler` for filtering out messages from recently revoked partitions.